app/src/main/java/org/mozilla/focus/compose/CFRPopupShape.kt

```diff
@@ -0,0 +1,179 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Outline
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+import org.mozilla.focus.R.color
+import org.mozilla.focus.ui.theme.FocusTheme
+import kotlin.math.roundToInt
+
+private const val INDICATOR_BASE_TO_HEIGHT_RATIO = 1.5f
+
+/**
+ * A [Shape] describing a popup with an indicator triangle shown at top.
+ *
+ * @param indicatorArrowStartOffset Distance between the popup start and the indicator arrow start
+ * @param indicatorArrowHeight Height of the indicator triangle. This influences the base length.
+ * @param cornerRadius The radius of the popup's corners.
+ * If [indicatorArrowStartOffset] is `0` then the top-start corner will not be rounded.
+ */
+class CFRPopupShape(
```

Comment:
I wonder if we should just call this by its full name instead of acronym - ideally we shouldn't expect people to have to know our acronyms. `ContextualFeatureRecommendationShape`,  `FeatureRecommendationShape`, ???

Reply:
Just a drive by thought, don't need another review from me.

Reply:
I do prefer explicative naming but in this case `ContextualFeatureRecommendation` seems just a tad long to be used all throughout the app. I see in Fenix `cfr` is used in many places and in tickets title / descriptions also so the entire feature might be named "CFR".